src/dash/DashHandler.js

```diff
@@ -352,8 +352,8 @@ function DashHandler(config) {
 
         index = getIndexForSegments(time, representation, timeThreshold);
         //Index may be -1 if getSegments needs to update.  So after getSegments is called and updated then try to get index again.
-        updateSegments(representation);
         if (index < 0) {
+            updateSegments(representation);
```

Comment:
This fixes a bug where when we pause let the player drift out of the rewind window, and then seek back to the live edge, the later getSegmentByIndex() call is happening on a different segment list to that used for getIndexForSegments(), - The effect being an index is chosen that's then no longer in the list. Is it valid to be updating the segment list in the middle of the function? Or valid even in this function entirely?